### PR TITLE
Lazy import dask.distributed to reduce import time of xarray

### PR DESCRIPTION
**Pull request by hmaarrfk**
I was auditing the import time of my software and found that distributed added a non insignificant amount of time to the import of xarray:

Using `tuna`, one can find that the following are sources of delay in import time for xarray:

To audit, one can use the the command
```
python -X importtime -c "import numpy as np; import pandas as pd; import dask.array; import xarray as xr" 2>import.log && tuna import.lo
```
The command as is, breaks out the import time of numpy, pandas, and dask.array to allow you to focus on "other" costs within xarray.
Main branch:
image

Proposed:
image

One would be tempted to think that this is due to xarray.testing and xarray.tutorial but those just move the imports one level down in tuna graphs.
image

- [x] ~~Closes~~
- [x] ~~Tests added~~
- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`
- [x] ~~New functions/methods are listed in `api.rst`~~

### CODE REVIEW

**Comment by headtr1ck** [comment]
Good first step towards https://github.com/pydata/xarray/issues/6726

File: xarray/backends/locks.py [hunk 1]
```diff
@@ -128,15 +131,12 @@ def acquire(lock, blocking=True):
     if blocking:
         # no arguments needed
         return lock.acquire()
-    elif DistributedLock is not None and isinstance(lock, DistributedLock):
-        # distributed.Lock doesn't support the blocking argument yet:
-        # https://github.com/dask/distributed/pull/2412
```
Comment (hmaarrfk) [inline xarray/backends/locks.py:133]: this seems to have been resolved in 2018.

**Review by Illviljan — APPROVED** [review]

**Comment by jhamman** [comment]
Thanks @hmaarrfk - this looks great.

**Comment by max-sixty** [comment]
Nice!!

> One would be tempted to think that this is due to xarray.testing and xarray.tutorial but those just move the imports one level down in tuna graphs.

Separate issue, but do these need to be imported into `xarray/__init__.py`? (I also can't see them in the third graph, but sounds like they're still there)

**Comment by hmaarrfk** [comment]
> Separate issue, but do these need to be imported into xarray/__init__.py

At this point removing testing and tutorial would be strange and break things. Stefan in the discussion linked above speaks about the reasoning behind importing submodules in the top level namespace.

**Comment by headtr1ck** [comment]
The auto labeler is backfiring a bit here, haha.

**Comment by dcherian** [comment]
```
--- this PR
2022-10-17T15:43:10.2509472Z [ 65.51%] ··· import.Import.timeraw_import_xarray                       818±30ms
2022-10-17T15:43:10.2510103Z [ 65.66%] ··· import.Import.timeraw_import_xarray_backends               816±3ms
2022-10-17T15:43:10.2510733Z [ 65.81%] ··· import.Import.timeraw_import_xarray_plot                   826±7ms

--- main
                              [90.51%] ··· import.Import.timeraw_import_xarray                     1.02±0.03s
2022-10-17T15:49:58.9125041Z [ 90.66%] ··· import.Import.timeraw_import_xarray_backends               974±7ms
2022-10-17T15:49:58.9125536Z [ 90.81%] ··· import.Import.timeraw_import_xarray_plot                  969±10ms
```

Nice improvement! Thanks @hmaarrfk

**Comment by Illviljan** [comment]
Yes. very nice! It's also fun that we can see the 20% improvement in both tuna and asv. :)

**Pull request merged**